pkg/server/server.go

```diff
@@ -141,6 +123,11 @@ func runControllers(ctx context.Context, config *Config) error {
 		if err := sc.Start(ctx); err != nil {
 			panic(err)
 		}
+		handler, err := router(controlConfig, controlConfig.Runtime.Tunnel, sc.Core.Core().V1().Secret())
+		if err != nil {
+			panic(errors.Wrap(err, "starting router"))
```

Comment:
Should we panic here or just return the error? The calling code will be calling log.Fatalf() on it.

Reply:
Sames goes for the untouched sc.Start() above

Reply:
Oh, I see. Its in the sub-goroutine. I hate this code.